Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -148,6 +148,13 @@ void finishDropInWithError(Exception e) {
         finish();
     }
 
+    private DropInRequest getDropInRequest(Intent intent) {
+        Bundle bundle = intent.getParcelableExtra(DropInClient.EXTRA_CHECKOUT_REQUEST_BUNDLE);
```

Comment:
This is awesome!

Reply:
My understanding of Intents was to pass data from one activity/fragment to another. I wonder if there's a potential case where the OS (or the merchant) clears Intents (or is doing something else with them), which in turn would break this flow.

Asking because if this isn't a common pattern - want to make there is no lost stability with this change.

Reply:
Yeah since Android calls the constructors directly behind the scenes the Intent gives us an opportunity to inject launch args for Activities/Fragments. They are for the most part kept in memory from my understanding, but if an activity / fragment is restored that process should be protected by the OS.

Reply:
Also to clarify - we were always passing the DropInRequest to DropInActivity through an intent extra, now we are just adding one more layer by wrapping it in a bundle, so that we can explicitly load it with the DropInRequest class loader